### PR TITLE
Add `ignore: ["custom-elements"]` to `selector-max-type`

### LINKED ISSUE #6579: Add `ignore: ["custom-elements"]` to `selector-max-type`
### What is the problem you're trying to solve?

My current config errors when a custom-element name is used:

```css
my-custom-element {
    /* Expected my-custom-element to have no more than 0 type selectors (selector-max-type) */
}
```

### What solution would you like to see?

I would like to be able to use 'custom-elements' ignore option as available in [selector-type-no-unknown](https://stylelint.io/user-guide/rules/selector-type-no-unknown/#ignore-custom-elements-default-namespace). Seems like re-using that may be beneficial?

### PR DESCRIPTION
**Pull request by muddv**
> Which issue, if any, is this issue related to?

- Closes #6579

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory

### CODE REVIEW

File: lib/rules/selector-max-type/__tests__/index.js [hunk 1]
```diff
@@ -489,6 +489,44 @@ testRule({
 	],
 });
 
+testRule({
+	//TODO
```
Comment (jeddy3) [inline lib/rules/selector-max-type/__tests__/index.js:493]:
```suggestion
```

File: lib/rules/selector-max-type/README.md [hunk 2]
```diff
@@ -151,6 +151,24 @@ div a + span {}
 #bar + div + span + a + span {}
 ```
 
+#### `"custom-elements"`
+
+Allow custom elements.
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md:156]:
```suggestion
Discount custom elements.

For example, with `2`:
```

File: lib/rules/selector-max-type/README.md [hunk 3]
```diff
@@ -77,7 +77,7 @@ div a .foo:not(span) {}
 
 ## Optional secondary options
 
-### `ignore: ["child", "compounded", "descendant", "next-sibling"]`
+### `ignore: ["child", "compounded", "descendant", "next-sibling", "custom-elements"]`
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md:80]:
```suggestion
### `ignore: ["child", "compounded", "custom-elements", "descendant", "next-sibling" ]`
```

Let's also move the section to be 3rd.

File: lib/rules/selector-max-type/README.md [hunk 4]
```diff
@@ -151,6 +151,24 @@ div a + span {}
...
+
+Allow custom elements.
+
+The following patterns are considered problems:
+
+<!-- prettier-ignore -->
+```css
+x-Foo {}
+```
+
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md:164]:
```suggestion
```

We only show patterns that are not problems for `ignore` options.

File: lib/rules/selector-max-type/README.md [hunk 5]
```diff
@@ -151,6 +151,24 @@ div a + span {}
...
+The following patterns are _not_ considered problems:
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md:165]:
```suggestion
The following pattern is _not_ considered a problem:
```

File: lib/rules/selector-max-type/README.md [hunk 6]
```diff
@@ -151,6 +151,24 @@ div a + span {}
...
+
+<!-- prettier-ignore -->
+```css
+x-foo {}
```
Comment (jeddy3) [inline lib/rules/selector-max-type/README.md:169]:
```suggestion
div a foo-bar {}
```

Let's tie it into the patterns outside of the `ignore` option.

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@muddv Thank you for contributing. It's looking good. I've made some suggestions to the docs to make them more consistent.

**Comment by muddv** [comment]
> @muddv Thank you for contributing. It's looking good. I've made some suggestions to the docs to make them more consistent.

Thank you for reviewing! I think I fixed everything, sorry for leaving some oversights in.

File: lib/rules/selector-max-type/README.md [hunk 7]
```diff
@@ -115,6 +115,19 @@ div span a.foo {}
 div span a#bar {}
 ```
 
+#### `"custom-elements"`
+
+Discount custom elements.
+
+For example, with `2`:
+
+The following patterns is _not_ considered a problem:
```
Comment (ybiquitous) [inline lib/rules/selector-max-type/README.md:124]: [trivial] Can you please fix the typo?

```suggestion
The following pattern is _not_ considered a problem:
```

**Review by ybiquitous — COMMENTED** [review]
@muddv Thanks for creating the PR. I point out a typo, but LGTM.

File: lib/rules/selector-max-type/README.md (hunk 7 above)
Comment (muddv) [reply lib/rules/selector-max-type/README.md]: Fixed now. Thanks for noticing, sorry  I did not catch it earlier

**Review by ybiquitous — APPROVED** [review]
Thanks. LGTM 👍🏼

**Review by jeddy3 — APPROVED** [review]
Thanks for making the changes. LGTM.

**Comment by muddv** [comment]
Thanks for reviewing!

**Pull request merged**